docs/developer/update-v1alpha1-provider-to-v1alpha2-infra-provider.md

```diff
@@ -0,0 +1,188 @@
+
+```golang
+// AWSClusterStatus defines the observed state of AWSCluster
+type AWSClusterStatus struct {
+    Ready   bool     `json:"ready"`
+    // APIEndpoints represents the endpoints to communicate with the control plane.
+    // +optional
+    APIEndpoints []APIEndpoint `json:"apiEndpoints,omitempty"`
+}
+```
+
+##### Provider cluster status `ready`
+
+A Provider`Cluster`'s `status` object must define a boolean field named `ready` and set the value to `true` only when the infrastructure required to provision a cluster is ready and available.
+
+##### Provider cluster status `apiEndpoints`
+
+A Provider`Cluster`'s `status` object may optionally define a field named `apiEndpoints` that is a list of the following objects:
+
+```golang
+// APIEndpoint represents a reachable Kubernetes API endpoint.
+type APIEndpoint struct {
+    // The hostname on which the API server is serving.
+    Host string `json:"host"`
+
+    // The port on which the API server is serving.
+    Port int `json:"port"`
+}
+```
+
+If present, this field is automatically inspected in order to obtain an endpoint at which the Kubernetes cluster may be accessed.
+
+## Create the infrastructure controllers
+
+The actuator model from v1alpha1 has been replaced by the infrastructure controllers in v1alpha2:
+
+| v1alpha1 | v1alpha2 |
+|---|---|
+| [`sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/cluster.Actuator`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e6a57dc61826b8c7806eba22a513c9722c420754/pkg/cloud/aws/actuators/cluster/actuator.go#L50-L57) | [`sigs.k8s.io/cluster-api-provider-aws/pkg/controller/awscluster.ReconcileAWSCluster`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L98-L103) |
+| [`sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.Actuator`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e6a57dc61826b8c7806eba22a513c9722c420754/pkg/cloud/aws/actuators/machine/actuator.go#L57-L65) | [`sigs.k8s.io/cluster-api-provider-aws/pkg/controller/awsmachine.ReconcileAWSMachine`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awsmachine/awsmachine_controller.go#L104-L109) |
+
+### The provider cluster controller
+
+Instead of processing the CAPI `Cluster` resources like the old actuator model, the new provider cluster controller operates on the new provider `Cluster` CRD. However, the overall workflow should feel the same as the old cluster actuator. For example, take the AWS cluster controller's [reconcile function](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L105-L162), it:
```

Comment:
This reads a bit confusing to me. instead of saying "new provider `Cluster` CRD", maybe say something along the lines of: "operates on a CRD that is defined by the infrastructure provider"?

Also, rather than "AWS cluster controller", it might be better to say "AWSCluster controller", so that it is clear that it is a controller for the `AWSCluster` type rather than a generic controller.